### DISCUSSION TITLE
Force spacing on SimpleGrid (CodeSandBox Example)

### DISCUSSION THREAD

**Discussion by ivanjeremic** [chakra-ui/chakra-ui#2049] [General] [answered]
I have a problem that I was not able to solve, The SimpleGrid responsive component is amazing but the spacing applies only correctly if there are enough items in the list, but in the real world sometimes we have fewer data and if there are only 3 items, for example, the grid items drift away from each other ignoring the spacing, what I want is that the items hold the size and go from left to right with correct spacing and be responsive, is there any way to solve this? I created a CodeSandBox with the problem.

[https://codesandbox.io/s/icy-moon-ku5ec?file=/src/App.js](https://codesandbox.io/s/simplegrid-problem-ku5ec)

**Comment by with-heart** [comment, marked as answer]
Hi @ivanjeremic! 

They're floating away from each other because you're setting `minChildWidth="120px"` on `SimpleGrid` while also setting `width="80px"` on each item.

The Chrome inspector allows us to see what's happening here:

![image](https://user-images.githubusercontent.com/1954752/93363586-7938f900-f815-11ea-8bb8-1a51bfcac6d0.png)

The first whitespace is the extra `40px` being added by `minChildWidth` along with a percent of the space remaining in the row, and the second whitespace is the `40px` gap. `minChildWidth` sets the grid to `auto-fit` items with `minmax(120px, 1fr)` meaning that items will be at minimum `120px` in width but stretch to fit any remaining space (minus the gaps between them).

If I understand what you're asking for, you should instead use `Grid`:

https://codesandbox.io/s/simplegrid-problem-forked-lfd5u?file=/src/App.js

In the top two grids, I've used `gridTemplateColumns="repeat(auto-fill, 80px)"` to tell the grid to fill as many _implicit columns_ of `80px` width into the available space as it can. This means you don't need to manually specify `width` on each item.

In the bottom two grids, I've also used `gridAutoRows="80px"` to tell it to automatically size _implicit rows_ with a height of `80px`, so you don't need to explicitly specify `width` or `height` on either item.

**Comment by arthurtg11** [comment]
Thank u very much
